### ISSUE TITLE
dfx deploy with gzip

### ISSUE THREAD

**Issue by lastmjs** [dfinity/sdk#2357]
Version `0.10.x` of `dfx` shipped with support for gzip Wasm binaries locally, but it doesn't work with `dfx deploy`. I would like to be able to simply change the path to the Wasm binary in my `dfx.json` and use the same process as without a gzipped binary

I would like to the following `dfx.json` file to be valid:

```json
{
    "canisters": {
        "bytes": {
            "type": "custom",
            "build": "npx azle bytes",
            "root": "src",
            "ts": "src/bytes.ts",
            "candid": "src/bytes.did",
            "wasm": "target/wasm32-unknown-unknown/release/bytes.wasm.gz",
            "declarations": {
                "output": "test/dfx_generated/bytes"
            }
        }
    }
}
```

**Comment by sesi200** [comment]
Agreed, there should be an easy way to do this.

**Comment by smallstepman** [comment]
fixed in https://github.com/dfinity/sdk/pull/2482
to be released in dfx 0.12.0

**Comment by gabrielnic** [comment]
I posted this question on the forum but maybe I can get an answer here as well https://forum.dfinity.org/t/dfx-deploy-with-gzip/16671 thanks
